system/modules/isotope/library/Isotope/Model/Payment/Saferpay.php

```diff
@@ -119,13 +119,18 @@ public function checkoutForm(IsotopeProductCollection $objOrder, \Module $objMod
         $objRequest->setHeader('Content-Type', 'application/x-www-form-urlencoded');
         $objRequest->send(static::createPayInitURI, http_build_query($this->generatePaymentPostData($objOrder), null, '&'), 'POST');
 
-        if ((int) $objRequest->code !== 200 || 0 !== strpos($objRequest->response, 'ERROR:')) {
+        if ((int) $objRequest->code !== 200 || 0 === strpos($objRequest->response, 'ERROR:')) {
             \System::log(sprintf('Could not get the redirect URI from Saferpay. See log files for further details.'), __METHOD__, TL_ERROR);
             log_message(sprintf('Could not get the redirect URI from Saferpay. Response was: "%s".', $objRequest->response), 'isotope_saferpay.log');
 
             Checkout::redirectToStep('failed');
         }
 
+        // redirect immediately
+        if (\Contao\Validator::isUrl($objRequest->response)) {
+            \Controller::redirect($objRequest->response);
```

Comment:
Can you explain what this is intended for? If the result was not a URL, the header field would not make sense either?